assume/units/building.py

```diff
@@ -0,0 +1,593 @@
+            technology (str, optional): Type of technology the building unit employs. Defaults to "building".
+            objective (str, optional): Optimization objective, e.g., "min_variable_cost". Defaults to "min_variable_cost".
+            node (str, optional): Network node where the building unit is connected. Defaults to "node0".
+            location (tuple[float, float], optional): Geographic coordinates (latitude, longitude) of the building. Defaults to (0.0, 0.0).
+            flexibility_measure (str, optional): Metric used to assess the building's flexibility. Defaults to "max_load_shift".
+            **kwargs: Additional keyword arguments for parent classes.
+
+        Raises:
+            ValueError: If any of the provided components are not recognized as valid technologies.
+            Exception: If none of the specified solvers are available.
+        """
+        super().__init__(
+            id=id,
+            unit_operator=unit_operator,
+            technology=technology,
+            components=components,
+            bidding_strategies=bidding_strategies,
+            index=index,
+            node=node,
+            location=location,
+            **kwargs,
+        )
+
+        # Validate provided components against the list of optional technologies
+        for component in self.components.keys():
+            if component not in self.optional_technologies:
+                raise ValueError(
+                    f"Component {component} is not a valid component for the building unit."
+                )
+
+        # Initialize forecasting data for various energy prices and demands
+        self.electricity_price = self.forecaster["price_EOM"]
+        self.natural_gas_price = self.forecaster["fuel_price_natural gas"]
+        self.heat_demand = self.forecaster["heat_demand"]
+        self.ev_load_profile = self.forecaster["ev_load_profile"]
+        self.battery_load_profile = self.forecaster["battery_load_profile"]
+        self.inflex_demand = self.forecaster[f"{self.id}_load_profile"]
+
+        self.objective = objective
+        self.flexibility_measure = flexibility_measure
+
+        # Flags indicating the presence of specific components
+        self.has_heatpump = "heat_pump" in self.components
+        self.has_boiler = "boiler" in self.components
+        self.has_thermal_storage = "thermal_storage" in self.components
+        self.has_ev = "electric_vehicle" in self.components
+        self.has_battery_storage = "generic_storage" in self.components
+        self.has_pv = "pv_plant" in self.components
+
+        # Configuration for electric vehicles selling energy to the market
+        if self.has_ev:
+            self.ev_sells_energy_to_market = self.components["electric_vehicle"].get(
+                "sells_energy_to_market", "true"
+            ).lower() in {"y", "yes", "t", "true", "on", "1"}
```

Comment:
Maybe these bool checks can be moved to a util to avoid code duplication

Reply:
yes true, you had a function for that, somehow it got lost, returned it back